### PR TITLE
[DNM][Distributed] Handle generic actors and Codable better

### PR DESCRIPTION
**Pull request by ktoso**
This builds on top of https://github.com/apple/swift/pull/71435

And adds the last commit which cleans up our "implicit codable" logic.
I think this is probably good enough tbh -- see test cases @xedin 

The commit: https://github.com/apple/swift/pull/71467/commits/5f669bb611dc63dcefcd4726813d34fdfca04960

### CODE REVIEW

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 1]
```diff
@@ -12,24 +12,25 @@
...
 #include "swift/AST/TypeCheckRequests.h"
-#include "swift/AST/NameLookupRequests.h"
-#include "swift/AST/ASTMangler.h"
-#include "swift/AST/DistributedDecl.h"
 #include "swift/Basic/Defer.h"
 #include "swift/ClangImporter/ClangModule.h"
 #include "swift/Sema/ConstraintSystem.h"
 #include "llvm/ADT/SmallString.h"
 #include "llvm/ADT/StringExtras.h"
-#include "DerivedConformances.h"
```
Comment (ktoso) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:32]: // clang tidy sorted it

**Comment by ktoso** [comment]
@swift-ci please smoke test

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 2]
```diff
@@ -763,6 +762,30 @@ addDistributedActorCodableConformance(
     return nullptr;
   }
 
+  if (actor->isGeneric()) {
```
Comment (slavapestov) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:764]: Do you want isGeneric() or isGenericContext()?

File: lib/Sema/CodeSynthesisDistributedActor.cpp [hunk 3]
```diff
@@ -763,6 +762,30 @@ addDistributedActorCodableConformance(
...
+      return nullptr;
+    }
+  }
+
+  assert(proto->isSpecificProtocol(swift::KnownProtocolKind::Decodable) ||
+         proto->isSpecificProtocol(swift::KnownProtocolKind::Encodable));
+
   // === Does the actor explicitly conform to the protocol already?
   auto explicitConformance =
       module->lookupConformance(actor->getInterfaceType(), proto);
```
Comment (slavapestov) [inline lib/Sema/CodeSynthesisDistributedActor.cpp:792]: This will always return false because getInterfaceType() returns a metatype given a TypeDecl

**Comment by ktoso** [comment]
Thanks Slava, I'll be looking into this one deeper next

**Pull request closed without merging**